frontend/nodefront/routes/courses.js

```diff
@@ -1,27 +1,82 @@
 var express = require('express');
 var router = express.Router();
-var https = require('https');
+var http = require('http');
 
 
-// https.get('http://localhost:8080/courses/listAll', (resp) => {
-//     let data = '';
-//
-//     resp.on('data', (chunk) => {
-//         data += chunk;
-//     });
-//
-//     resp.on('end', () => {
-//         console.log(JSON.parse(data).explanation);
-//     });
-//
-// }).on("error", (err) => {
-//     console.log("Error: " + err.message);
-// });
+function api_get_course(subject, catalog) {
+
+    return new Promise((resolve, reject) => {
+        http.get('http://localhost:8080/courses/4417', (res) => {
```

Comment:
harcoded but will just do like http://localhost:8080/courses/{subject}/{catalog} if that endpoint exists